Compatibility Issue with Axum 0.7.1 and shuttle_axum

## Summary

After updating `axum` to version 0.7.1 the compiler throws an error related to a trait bound not being satisfied between axum::Router and shuttle_axum::AxumService.

## Steps to Reproduce

1. Run `cargo shuttle init --template axum`.
2. Update `axum` from 0.6.20 to 0.7.1 in `Cargo.toml`.
3. Attempt to compile the project.
4. Observe the following compilation error:

```
error[E0277]: the trait bound `shuttle_axum::AxumService: std::convert::From<axum::Router<_>>` is not satisfied
  --> src/main.rs:11:15
   |
11 |     Ok(router.into())
   |               ^^^^ the trait `std::convert::From<axum::Router<_>>` is not implemented for `shuttle_axum::AxumService`
   |
   = help: the trait `std::convert::From<axum::routing::Router>` is implemented for `shuttle_axum::AxumService`
   = note: required for `axum::Router<_>` to implement `std::convert::Into<shuttle_axum::AxumService>`
```

--- a/services/shuttle-axum/Cargo.toml
+++ b/services/shuttle-axum/Cargo.toml
@@ -9,8 +9,14 @@ keywords = ["shuttle-service", "axum"]
 [workspace]
 
 [dependencies]
-axum = { version = "0.6.10" }
+axum = { version = "0.6.13", optional = true }
+axum-0-7 = { package = "axum", version = "0.7.1", optional = true }
 shuttle-runtime = { path = "../../runtime", version = "0.34.1", default-features = false }
 
 [dev-dependencies]
 tokio = { version = "1.26.0", features = ["macros", "rt-multi-thread"] }
+
+[features]
+default = ["axum"]
+
+axum-0-7 = ["dep:axum-0-7"]

--- a/services/shuttle-axum/src/lib.rs
+++ b/services/shuttle-axum/src/lib.rs
@@ -19,25 +19,40 @@
 use shuttle_runtime::{CustomError, Error};
 use std::net::SocketAddr;
 
+#[cfg(feature = "axum")]
+use axum::Router;
+#[cfg(feature = "axum-0-7")]
+use axum_0_7::Router;
+
 /// A wrapper type for [axum::Router] so we can implement [shuttle_runtime::Service] for it.
-pub struct AxumService(pub axum::Router);
+pub struct AxumService(pub Router);
 
 #[shuttle_runtime::async_trait]
 impl shuttle_runtime::Service for AxumService {
     /// Takes the router that is returned by the user in their [shuttle_runtime::main] function
     /// and binds to an address passed in by shuttle.
     async fn bind(mut self, addr: SocketAddr) -> Result<(), Error> {
+        #[cfg(feature = "axum")]
         axum::Server::bind(&addr)
             .serve(self.0.into_make_service())
             .await
             .map_err(CustomError::new)?;
+        #[cfg(feature = "axum-0-7")]
+        axum_0_7::serve(
+            shuttle_runtime::tokio::net::TcpListener::bind(addr)
+                .await
+                .map_err(CustomError::new)?,
+            self.0,
+        )
+        .await
+        .map_err(CustomError::new)?;
 
         Ok(())
     }
 }
 
-impl From<axum::Router> for AxumService {
-    fn from(router: axum::Router) -> Self {
+impl From<Router> for AxumService {
+    fn from(router: Router) -> Self {
         Self(router)
     }
 }

